xdem/coreg/base.py

```diff
@@ -1224,18 +1528,52 @@ class Coreg:
 
     _fit_called: bool = False  # Flag to check if the .fit() method has been called.
     _is_affine: bool | None = None
+    _is_translation: bool | None = None
     _needs_vars: bool = False
     _meta: CoregDict
 
-    def __init__(self, meta: CoregDict | None = None) -> None:
+    def __init__(self, meta: dict[str, Any] | None = None) -> None:
         """Instantiate a generic processing step method."""
-        self._meta: CoregDict = meta or {}  # All __init__ functions should instantiate an empty dict.
+
+        # Automatically sort input keys into their appropriate nested level using only the TypedDicts defined
+        # above which make up the CoregDict altogether
+        dict_meta = CoregDict(inputs={}, outputs={})
+        if meta is not None:
+            # First, we get the levels ("random", "fitorbin", etc)
+            list_input_levels = list(InputCoregDict.__annotations__.keys())
+            # Then the list of keys per level
+            keys_per_level = [
+                list(globals()[InputCoregDict.__annotations__[lv].__forward_arg__].__annotations__.keys())
```

Comment:
similar question about "__forward_arg__".

Reply:
Also added a comment here. This is to get the value of a Typed Dict class key (so get the nested dict class).